packages/flutter/lib/src/rendering/table.dart

```diff
@@ -0,0 +1,784 @@
+      verticalInside: BorderSide.lerp(a.verticalInside, b.verticalInside, t)
+    );
+  }
+
+  @override
+  bool operator ==(dynamic other) {
+    if (super != other)
+      return false;
+    final TableBorder typedOther = other;
+    return horizontalInside == typedOther.horizontalInside &&
+           verticalInside == typedOther.verticalInside;
+  }
+
+  @override
+  int get hashCode => hashValues(super.hashCode, horizontalInside, verticalInside);
+
+  @override
+  String toString() => 'TableBorder($top, $right, $bottom, $left, $horizontalInside, $verticalInside)';
+}
+
+/// Vertical alignment options for cells in [RenderTable] objects.
+///
+/// This is specified using [TableCellParentData] objects on the
+/// [RenderObject.parentData] of the children of the [RenderTable].
+enum CellVerticalAlignment {
+  /// Cells with this alignment are placed with their top at the top of the row.
+  top,
+
+  /// Cells with this alignment are vertically centered in the row.
+  middle,
+
+  /// Cells with this alignment are placed with their bottom at the bottom of the row.
+  bottom,
+
+  /// Cells with this alignment are aligned such that they all share the same
+  /// baseline. Cells with no baseline are top-aligned instead. The baseline
+  /// used is specified by [RenderTable.baseline]. It is not valid to use the
+  /// baseline value if [RenderTable.baseline] is not specified.
+  baseline,
+
+  /// Cells with this alignment are sized to be as tall as the row, then made to fit the row.
+  /// If all the cells have this alignment, then the row will have zero height.
+  fill
+}
+
+class RenderTable extends RenderBox {
+  RenderTable({
+    int columns,
+    int rows,
+    Map<int, ColumnWidth> columnWidths,
+    ColumnWidth defaultColumnWidth: const FlexColumnWidth(1.0),
+    TableBorder border,
+    CellVerticalAlignment defaultVerticalAlignment: CellVerticalAlignment.top,
+    TextBaseline textBaseline,
+    List<List<RenderBox>> children
+  }) {
+    assert(columns != null || (children != null && children.isNotEmpty));
+    assert(columns == null || columns >= 0);
+    assert(rows == null || rows >= 0);
+    assert(rows == null || children == null);
+    assert(defaultColumnWidth != null);
+    _columns = columns ?? children.first.length;
+    _rows = rows ?? 0;
+    _children = new List<RenderBox>()..length = _columns * _rows;
+    _columnWidths = columnWidths ?? new HashMap<int, ColumnWidth>();
+    _defaultColumnWidth = defaultColumnWidth;
+    _border = border;
+    _defaultVerticalAlignment = defaultVerticalAlignment;
+    _textBaseline = textBaseline;
+    for (List<RenderBox> row in children)
+      addRow(row);
+  }
+
+  // TODO(ianh): Add a 'decoration' field to the children's parent data, to paint on each cell.
+
+  // Children are stored in row-major order.
+  // _children.length must be rows * columns
+  List<RenderBox> _children = const <RenderBox>[];
+
+  int get columns => _columns;
+  int _columns;
+  void set columns(int value) {
```

Comment:
You'll probably want a setter that sets the number of rows, columns, and children all at once.  Without that, there's no efficient want to add a column to the table.